src/app/zap-templates/templates/chip/helper.js

```diff
@@ -445,6 +446,127 @@ async function chip_endpoint_clusters(options)
   return templateUtil.collectBlocks(clusters, options, this);
 }
 
+/**
+ * Helper checks if the type for the bitmap is Bitflags. This generally includes
+ * all inherited bitmaps i.e. all bitmaps apart from
+ * bitmap8/16/32(generally defined in types.xml)
+ * example:
+ * {{#if_is_strongly_typed_bitmap type}}
+ * strongly typed bitmap
+ * {{else}}
+ * not a strongly typed bitmap
+ * {{/if_is_strongly_typed_bitmap}}
+ *
+ * @param {*} type
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_bitmap(type, options)
+{
+  let packageId = await templateUtil.ensureZclPackageId(this)
+  let bitmap
+  if (type && typeof type === 'string')
+  {
+    bitmap = await queryZcl.selectBitmapByName(this.global.db, packageId, type)
+  }
+  else
+  {
+    bitmap = await queryZcl.selectBitmapById(this.global.db, type)
+  }
+
+  if (bitmap) {
+    let a = await queryZcl.selectAtomicType(this.global.db, packageId, bitmap.name)
+    if (a)
+    {
+      return options.inverse(this)
+    }
+    else
+    {
+      return options.fn(this)
+    }
+  }
+  return options.inverse(this)
+}
+
+/**
+ * This function retrieves all the strongly typed enums from the database. This
+ * generally includes all inherited enums i.e. all enums apart from
+ * enum8/16/32(generally defined in types.xml)
+ *
+ * * example for if_is_strongly_typed_enum:
+ * {{#if_is_strongly_typed_enum type}}
+ * strongly typed enum
+ * {{else}}
+ * not a strongly typed enum
+ * {{/if_is_strongly_typed_enum}}
+ *
+ * @param {*} type
+ * @param {*} options
+ * @param {*} context
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_enum_common(type, options, context)
+{
+  let packageId = await templateUtil.ensureZclPackageId(context)
+  let enumRes
+  if (type && typeof type === 'string')
+  {
+    enumRes = await queryZcl.selectEnumByName(context.global.db, type, packageId)
+  }
+  else
+  {
+    enumRes = await queryZcl.selectEnumById(context.global.db, type)
+  }
+
+  if (enumRes) {
+    let a = await queryZcl.selectAtomicType(context.global.db, packageId, enumRes.name)
+    if (a)
+    {
+      return options.inverse(context)
+    }
+    else
+    {
+      return options.fn(context)
+    }
+  }
+  return options.inverse(context)
+}
+
+/**
+ * Helper that checks if an enum is a strongly typed enum or not
+ * This generally includes all inherited enums i.e. all enums apart from
+ * enum8/16/32(generally defined in types.xml)
+ *
+ * * example:
+ * {{#if_is_strongly_typed_enum type}}
+ * strongly typed enum
+ * {{else}}
+ * not a strongly typed enum
+ * {{/if_is_strongly_typed_enum}}
+ *
+ * @param {*} type
+ * @returns Promise of content.
+ */
+async function if_is_strongly_typed_enum(type, options)
```

Comment:
This is used for the special case of vendor_id. Just like If_chip_enum before.

Reply:
This is used in `src/darwin/Framework/CHIP/templates/partials/attribute_data_callback_name.zapt`

Reply:
It shouldn't be used there.  That code can just use `if_is_strongly_typed_chip_enum`, because it has already checked for the vendor_id special-case.  Again, we should just not have this function, should not have the "common" helper, and `if_is_strongly_typed_chip_enum` should be the only enum-related thing here.